alica_engine/include/engine/blackboard/BlackBoard.h

```diff
@@ -15,49 +12,25 @@ namespace alica
 class BlackBoard
 {
 public:
-    using IdType = BBIdent;
-    using ObjectType = ByteArray;
     BlackBoard() = default;
 
-    IdType registerValue(const int8_t* buffer, int len);
-    IdType registerValue(const char* buffer, int len);
-
-    template <class InputIt>
-    IdType registerValue(InputIt begin, InputIt end);
-
-    bool hasValue(IdType id) const { return _body.find(id) != _body.end(); }
+    void registerValue(const std::string& key, std::any any);
+    void registerValue(const char* key, std::any any);
 
-    const ObjectType& getValue(IdType id) const { return _body.find(id)->second; }
-    void removeValue(IdType id) { _body.erase(_body.find(id)); }
-
-    bool empty() const { return _body.empty(); }
-    int size() const { return _body.size(); }
+    template <typename T> const T getValue(const std::string& key) const {
```

Comment:
returning by const value is pretty unnecessary & it prevents moving from the return value. Best to just return T & also any_cast to T.

Reply:
Can't return non const ref due to race conditions.  Probably I'll just make it so all access is through an RAII lock object since we'll eventually need that anyway to be able to edit.

Reply:
I meant return `T` (i.e. by value) and do `any_cast<T>`